adr/ADR-0009-project-workflow.md

```diff
@@ -0,0 +1,623 @@
+```mermaid
+flowchart TD
+    UM[upstream:main] --> |feature PRs| UM
+    UM --> |community releases| UT[tags: v*-community]
+    UT --> |automated build| UI[upstream:v*-community]
+    UM --> |automated daily sync| MM[midstream:main]
+    MM --> |feature PRs| MI[midstream:incubation]
+    MI --> |periodic ODH releases| MOB[midstream:odh/x.y]
+    MOB --> |tag| MT[v*-odh]
+    MT --> |automated build| MOI[midstream:v*-odh]
+    MI --> |cherry-pick approved features| MS[midstream:stable]
+    MS --> |automated sync| DB[downstream:main]
+    DB --> |monthly release| DR[downstream:release-x]
+    DR --> |tag| DT[v*]
+
+    subgraph "Docker Image Updates"
+    UM -.-> |update params.env| OP[operator params]
+    MOI -.-> |update odh overlay| OP
+    DT -.-> |update downstream overlay| OP
+    end
+```
+
+### Commit Lifecycle Diagrams
+
+The following diagrams illustrate the specific lifecycle of commits through the system:
+
+> **Legend**:
+> - 🤖 Automated process
+> - 👤 Manual/human process
+
+#### Upstream Feature → Incubation → Stable → Downstream
+
+```mermaid
+sequenceDiagram
+    participant U as Upstream:main
+    participant MM as Midstream:main
+    participant MI as Midstream:incubation
+    participant MOB as Midstream:odh/x.y
+    participant MS as Midstream:stable
+    participant D as Downstream:main
+
+    Note over U: Feature development
+    U->>U: 👤 Feature commit merged
+    Note over U,MM: Daily automated sync
+    U->>MM: 🤖 Feature synced to midstream:main
+    Note over MM,MI: Feature testing
+    MM->>MI: 👤 Feature promoted to incubation
+    Note over MI,MOB: Semi-production testing
+    MI->>MOB: 🤖 Cut release branch odh/x.y for testing
+    Note over MOB: Tag & Build
+    MOB->>MOB: 🤖 Tag v*-odh, build images for ODH
+    Note over MI,MS: Validated feature
+    MI->>MS: 👤 Cherry-pick validated features to stable
+    Note over MS,D: Downstream propagation
+    MS->>D: 🤖 Automated sync to downstream
+    D->>D: 🤖 Downstream release
+```
+
+#### Feature Incubation Process
+
+```mermaid
+sequenceDiagram
+    participant U as Upstream:main
+    participant MM as Midstream:main
+    participant MI as Midstream:incubation
+    participant MOB as Midstream:odh/x.y
+    participant MS as Midstream:stable
+
+    Note over U,MM: Features arrive from upstream
+    U->>MM: 🤖 Daily sync brings in new features
+    Note over MM,MI: Feature incubation
+    MM->>MI: 👤 Select features moved to incubation
+    MI->>MI: 👤+🤖 Integration testing with other features
+    Note over MI,MOB: Semi-production validation
+    MI->>MOB: 🤖 Cut release branch for ODH testing
+    MOB->>MOB: 🤖 Build images for ODH deployment
+    MOB->>MOB: 👤+🤖 Test in semi-production environments
+    Note over MI,MS: Feature curation
+    MI->>MS: 👤 Validated features promoted to stable
+    Note over MI: Feature validation cycle
+    MI->>MI: 👤 Some features may require more testing
+    MI->>MS: 👤 Later promotion when validated
+```
+
+#### Critical Security Fix Path
```

Comment:
1. is this only for Trivy result or downstream Synk too? they might show different sevirity.  
2. do we follow the same process for downstream "blocker" fix?